client/src/pages/Analytics/components/CategoryGraph.tsx

```diff
@@ -17,24 +21,85 @@ const categoryTotals = [
  * @returns {JSX.Element} - category graph
  */
 export const CategoryGraph: React.FC = () => {
+  const { userId, token } = useUser();
+  const [categoryTotals, setCategoryTotals] = React.useState<
+    { id: number; value: number; label: string }[]
+  >([]);
+  const [errorState, setErrorState] = React.useState(false);
+  const currDate = new Date();
+  const startDate = new Date(currDate.getFullYear(), currDate.getMonth(), 1);
+
+  React.useEffect(() => {
+    fetchCategoryExpenses(startDate.toISOString(), currDate.toISOString());
+  }, []);
+
+  const fetchCategoryExpenses = async (startDate: string, endDate: string) => {
+    try {
+      if (userId != null && token != null) {
+        const response = await categoryTotalExpensesRequest(
+          { userId, token },
+          { startDate, endDate },
+        );
+
+        if (response.status === 200) {
+          const totals: { category: string; amount: number }[] =
+            response.data.data;
+          const processedTotals = processData(totals);
+          setCategoryTotals(processedTotals);
+        }
+      }
+    } catch (error) {
+      setErrorState(true);
+    }
+  };
+
+  const processData = (
+    totals: { category: string; amount: number }[],
+  ): { id: number; value: number; label: string }[] => {
+    const categoryMap = new Map(
+      totals.map((item) => [item.category, item.amount]),
+    );
+    const processedData = allCategories.map((category, index) => ({
+      id: index,
+      value: categoryMap.get(category) || 0,
```

Comment:
if you can make the process data value using the `capitalizeWord` from the utils, its on my group split ui, just so that it doesnt show all caps in the ui, I can fix that during cleanup